src/main/kotlin/com/github/jyoo980/reachhover/listeners/EditorHoverListener.kt

```diff
@@ -0,0 +1,19 @@
+package com.github.jyoo980.reachhover.listeners
+
+import com.github.jyoo980.reachhover.util.MouseHoverUtil
+import com.intellij.openapi.diagnostic.Logger
+import com.intellij.openapi.editor.event.EditorMouseEvent
+import com.intellij.openapi.editor.event.EditorMouseMotionListener
+
+internal class EditorHoverListener : EditorMouseMotionListener {
+
+    private val logger: Logger = Logger.getInstance(EditorHoverListener::class.java)
+
+    override fun mouseMoved(e: EditorMouseEvent) {
+        val offset = MouseHoverUtil.targetOffset(e) ?: return
+        val project = e.editor.project ?: return
```

Comment:
I could do something like 
```
MouseHoverUtil.targetOffset(e)?.let { offset ->
  e.editor.project?.let { project ->
    ...
}
```
But I feel like that'll get pretty messy if we end up having more than 2 nullables.

I know in Scala you could have for-comprehensions, like

```scala
for {
  a <- maybeA
  b <- maybeB
} yield ...
```
But I'm not sure if there's anything like that in Kotlin.

Reply:
I don't know such a way as well. But it doesn't mean that it doesn't exist :) 
I think the way with `?: return` is just fine.